rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/docker/DockerImageNameFactory.groovy

```diff
@@ -0,0 +1,24 @@
+package com.netflix.spinnaker.rosco.providers.docker
+
+import com.netflix.spinnaker.rosco.api.BakeRequest
+import com.netflix.spinnaker.rosco.providers.util.ImageNameFactory
+import com.netflix.spinnaker.rosco.providers.util.PackageNameConverter
+
+class DockerImageNameFactory extends ImageNameFactory {
+
+  @Override
+  def buildAppVersionStr(BakeRequest bakeRequest, List<PackageNameConverter.OsPackageName> osPackageNames) {
+    super.buildAppVersionStr(bakeRequest, osPackageNames) ?: clock.millis().toString()
+  }
+
+  @Override
+  def buildImageName(BakeRequest bakeRequest, List<PackageNameConverter.OsPackageName> osPackageNames) {
+
+    String imageName = bakeRequest.ami_name ?: osPackageNames.first()?.name
+    String imageNamePrefixed = [bakeRequest.organization, imageName].findAll({it}).join("/")
+    String imageNameSuffixed = [imageNamePrefixed, bakeRequest.ami_suffix].findAll({it}).join("-")
+
+    return imageNameSuffixed
```

Comment:
If you mean the ```docker_image_tag``` the buildAppVersionStr is used by default in the DockerBakeHandler --> https://github.com/brujoand/rosco/blob/88a2bbce08caeb3d6c095adb6b904a8bd45713ec/rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/docker/DockerBakeHandler.groovy#L74 

I'm thinking that this would keep valuable information tagged to the image, and a next step could be to add support for multiple tags pr image from a user provided list. But as is, one could ofc put ``` docker_target_image_tag ``` in the extended attributes and override it.